src/Stream.ts

```diff
@@ -1,23 +1,25 @@
 import { UnsubscribeFn } from './ApiClient'
-import Message from './Message'
+import { MessageV1, MessageV2 } from './Message'
 import Client from './Client'
 import { messageApi, fetcher } from '@xmtp/proto'
 
-export type MessageTransformer<T> = (msg: Message) => T
+export type MessageDecoder<M> = (env: messageApi.Envelope) => Promise<M>
 
-export type MessageFilter = (msg: Message) => boolean
+export type MessageTransformer<M, T> = (msg: M) => T
 
-export type ContentTopicUpdater = (msg: Message) => string[] | undefined
+export type MessageFilter<M> = (msg: M) => boolean
 
-export const noTransformation = (msg: Message): Message => {
+export type ContentTopicUpdater<M> = (msg: M) => string[] | undefined
+
+export const noTransformation = (msg: any) => {
```

Comment:
Not sure how to avoid the any here, to provide an identity function for both versions.